modules/afsocket/socket-options-inet.c

```diff
@@ -166,6 +198,26 @@ socket_options_inet_setup_socket(SocketOptions *s, gint fd, GSockAddr *addr, AFS
   return TRUE;
 }
 
+void
+socket_options_inet_set_interface_name(SocketOptionsInet *self, const gchar *interface_name)
+{
+  if (self->interface_name)
+    {
+      g_free(self->interface_name);
+    }
+  self->interface_name = g_strdup(interface_name);
+}
+
+void
+socket_options_inet_free(gpointer s)
+{
+  SocketOptionsInet *self = (SocketOptionsInet *) s;
+  if (self->interface_name)
+    {
+      g_free(self->interface_name);
+    }
+}
```

Comment:
`g_free(s)` is missing from here. There is no `socket_options_free_method()`, so I guess we have to add it or just use `g_free(s)` directly.